File: programmers/Level 2/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(string a, string b){
    return (a + b > b + a);
}

string solution(vector<int> numbers) {
    string answer = "";

    vector<string> v;
    for(auto number: numbers)
        v.push_back(to_string(number));

    sort(v.begin(), v.end(), compare);
    if(v[0] == "0")
        return "0";

    for(auto str: v){
        answer += str;
    }

    return answer;
}

File: programmers/Level 2/가장 큰 수.py
from functools import cmp_to_key


def comparator(a, b):
    if int(a + b) > int(b + a):
        return -1
    else:
        return 1


def solution(numbers):
    answer = ''
    numbers = [str(n) for n in numbers]
    numbers.sort(key=cmp_to_key(comparator))

    for s in numbers:
        answer += s

    return str(int(answer)) # 000 -> 0 위해 str(int())